From: Mathieu Eyraud <70028899+meyraud705@users.noreply.github.com>
Date: Sun, 14 Aug 2022 14:55:32 +0200
Subject: Add SDL_JOYBATTERYUPDATED event to SDL_JoystickEventState()

Origin: upstream, 2.23.3, commit:c6c688ab0184234f3ea8240be80df405fdda5b9a
---
 src/joystick/SDL_joystick.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index 94217cc..26b017b 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1743,7 +1743,8 @@ SDL_JoystickEventState(int state)
 #else
     const Uint32 event_list[] = {
         SDL_JOYAXISMOTION, SDL_JOYBALLMOTION, SDL_JOYHATMOTION,
-        SDL_JOYBUTTONDOWN, SDL_JOYBUTTONUP, SDL_JOYDEVICEADDED, SDL_JOYDEVICEREMOVED
+        SDL_JOYBUTTONDOWN, SDL_JOYBUTTONUP, SDL_JOYDEVICEADDED, SDL_JOYDEVICEREMOVED,
+        SDL_JOYBATTERYUPDATED
     };
     unsigned int i;
 
