From: Cameron Cawley <ccawley2011@gmail.com>
Date: Thu, 8 Sep 2022 15:53:25 +0100
Subject: Fix SDL_PIXELFORMAT_INDEX1LSB test case

Origin: upstream, 2.24.1, commit:59e04dcc0b2d3bb63678a4bd67fea9c5f579608a
---
 test/testautomation_surface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/testautomation_surface.c b/test/testautomation_surface.c
index 63d4c29..9b38066 100644
--- a/test/testautomation_surface.c
+++ b/test/testautomation_surface.c
@@ -693,7 +693,7 @@ surface_testOverflow(void *arg)
     SDLTest_AssertCheck(surface != NULL, "17px * 1 bit per px fits in 3 bytes: %s",
                         surface != NULL ? "(success)" : SDL_GetError());
     SDL_FreeSurface(surface);
-    surface = SDL_CreateRGBSurfaceFrom(buf, 7, 1, 1, 3, 0, 0, 0, 0);
+    surface = SDL_CreateRGBSurfaceFrom(buf, 17, 1, 1, 3, 0, 0, 0, 0);
     SDLTest_AssertCheck(surface != NULL, "17px * 1 bit per px fits in 3 bytes: %s",
                         surface != NULL ? "(success)" : SDL_GetError());
     SDL_FreeSurface(surface);
