From: Simon McVittie <smcv@debian.org>
Date: Mon, 30 May 2022 16:25:47 +0100
Subject: testatomic: Skip on Ubuntu aarch64

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libsdl2/+bug/1976288
Forwarded: not-needed
---
 test/testatomic.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/testatomic.c b/test/testatomic.c
index 7dbad45..99efb38 100644
--- a/test/testatomic.c
+++ b/test/testatomic.c
@@ -708,6 +708,9 @@ main(int argc, char *argv[])
     /* Enable standard application logging */
     SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
 
+#if defined(__aarch64__) && defined(UBUNTU_WORKAROUND)
+    SDL_Log("SKIP https://bugs.launchpad.net/ubuntu/+source/libsdl2/+bug/1976288");
+#else
     RunBasicTest();
 
     if (SDL_getenv("SDL_TESTS_QUICK") != NULL) {
@@ -721,6 +724,7 @@ main(int argc, char *argv[])
     RunFIFOTest(SDL_FALSE);
 #endif
     RunFIFOTest(SDL_TRUE);
+#endif
     return 0;
 }
 
