From: Cameron Gutman <aicommander@gmail.com>
Date: Tue, 16 Aug 2022 19:50:59 -0500
Subject: testgles2: Fix typo in help text

Origin: upstream, 2.23.3, commit:99e9156ff59825f093da11b93c92e2f562752d04
---
 test/testgles2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/testgles2.c b/test/testgles2.c
index 6ef805b..b5c0ec8 100644
--- a/test/testgles2.c
+++ b/test/testgles2.c
@@ -583,7 +583,7 @@ main(int argc, char *argv[])
             }
         }
         if (consumed < 0) {
-            static const char *options[] = { "[--fsaa]", "[--accel]", "[--zdepth %d], [--threaded]", NULL };
+            static const char *options[] = { "[--fsaa]", "[--accel]", "[--zdepth %d]", "[--threaded]", NULL };
             SDLTest_CommonLogUsage(state, argv[0], options);
             quit(1);
         }
